Maintenance/migrate docs

bthompson0 opened the pull request.

aheritier commented:
ouch @bthompson0 I started to do the same but with Asciidoc: https://github.com/aheritier/pipeline-maven-plugin/commit/f16c262f5795f19ae7a37ca2315861711e1ce6f2
But I just started I didn't finish (I had to work on the security related issues first) thus we can take yours.
I choose ASCIIDOC which is a bit more powerful than MD but for such docs ... I don't think it has a big importance.
I'll review your PR later this week. Thanks for your help

bthompson0 commented:
Hmmm ok.  Knowing that the team likes to use Asciidoc is a big help.  I'll use that in future PRs.

aheritier commented:
The jenkins community uses both.
I regularly use Markdown in others plugins. When I saw the size complexity of the page I prefered to use Asciidoc which is offering more features to format the page (document the code samples ..)
I am still not sure what the rendering is giving in the plugins application side.

bthompson0 commented:
Thanks for the thorough response.  I'm going to be more active on Github and contributing to Jenkins projects now.  I have a lot of time that freed up.  Sorry for opening the PR and letting it sit and not responding quickly.

> For now I didn't have the time to review your PR but I hope to do it next week (I should take some days off and could spend some hours on this plugin).

No worries, let me make the changes to convert to Asciidoc first.  Should be ready by tomorrow.

> I wanted to be sure that we grab the images from the wik

Thanks for attaching the files.  Just DLed them.

> I just updated my branch to have a better readibility on what I did

I may just work off of your branch on my fork if you think that would be better?  That way we both don't lose our work.

aheritier commented:
Thanks a lot @bthompson0 
For sure do not hesitate to reuse my branch if it's easier.

bthompson0 commented:
Still not completely ready.  Need to link the new images and cleanup some other existing links.  Also going to run a beautifier on it.

aheritier commented:
It's already really good. thanks a lot @bthompson0 
I am not sure that you can change the labels (I think you have to be developer) but you can perhaps downgrade the PR to draft while you are working on it.
Anyway, just ping me when you want to have it reviewed merged.
I hope that we extracted all images from confluence. If it's not the case, let me know and I'll find a solution to get them.

About the FAQ (not sure if you want to convert it to asciidoc too) but I noticed that we have 2 references in the codebase:
* https://github.com/jenkinsci/pipeline-maven-plugin/blob/master/jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/MavenSpyLogProcessor.java#L207 is a reference to the confluence page
* https://github.com/jenkinsci/pipeline-maven-plugin/blob/master/jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution2.java#L195 is a reference to another FAQ ( ???) https://github.com/jenkinsci/pipeline-maven-plugin/blob/master/jenkins-plugin/src/resources/faq.md

Both entries should be updated to the new FAQ. We should probably merge `jenkins-plugin/src/resources/faq.md` in the new FAQ and replace it's content to a note with the new link to use (otherwise old releases of the plugin will have a 404)

Thanks again for all this work

bthompson0 commented:
@aheritier Thanks, and those are great suggestions!  I made changes accordingly.

- All images should now be using the repository's images, not links anymore (there is 1 exception).
- Linked to Github repositories for Jenkins plugins where I could.  I couldn't find one of them (Task Scanner Plugin).
- The `jenkins-plugins/src/resources/faq.md` was already merged into the main faq.md so I went ahead and just converted it over to Asciidoc.
- Did some very minor formatting fixes (didn't try to get VSCode autoformat/Prettier to work for Asciidoc yet, can discuss adding a .prettierrc later if you'd like).
- Updated the links in the 2 Java source files.

That pretty much sums it up.  The exception to the images is that I couldn't find the warning.svg image from the link and I didn't have it on my local FS (I don't think).  Maybe I just forgot to upload it.  I'll double-check.  If you see another commit that's what it's for.

Ready for review, no rush.

aheritier commented on README.adoc:
```diff
@@ -0,0 +1,471 @@
+against the workspace root and not the current working directory. +
+ie. `$WORKSPACE/.repository` if `.repository` value is specified.
+
+image:docs/images/information.svg[(info)]
+`mavenSettingsConfig` and `globalMavenSettingsConfig` use the *ID*,
+not the *name*, of the Maven settings file (resp Maven Global Settings
+file).
```
```suggestion
NOTE: `mavenSettingsConfig` and `globalMavenSettingsConfig` use the *ID*, not the *name*, of the Maven settings file (resp Maven Global Settings file).
```

Use Admonition

aheritier commented on README.adoc:
```diff
@@ -0,0 +1,471 @@
+is installed)
+* Concordion test reports (since 3.0.0)
+
+image:docs/images/information.svg[(info)]
+The detection of Maven builds require to use Maven 3.2+.
```
```suggestion
NOTE: The detection of Maven builds require to use Maven 3.2+.
```

Use Admonition

aheritier commented on README.adoc:
```diff
@@ -0,0 +1,471 @@
+Trigger downstream pipeline that depend on Maven artifact generated by
+upstream pipelines.
+
+image:docs/images/information.svg[(info)] Notes
+
+* The upstream artifact must be generated in a "[.code]``withMaven(){}``"
+wrapping step to be detected by the triggering system
+* The downstream pipeline must have selected the build trigger "Build
+whenever a SNAPSHOT dependency is built"
+ ** The build trigger can be defined at the pipeline level ("Build
+Triggers"), at the multibranch pipeline level ("Scan Repository
+Triggers") or at the GitHub Organization / Bitbucket Project
+level ("Scan Organizations Triggers")
+* You have to manually trigger once the upstream pipeline and the
+downstream pipeline so that the link between the pipelines based on
+the SNAPSHOT dependency is established
+* The dependency graph is, for the moment, exclusively stored in an H2
+embedded database
+("[.code]``$JENKINS_HOME/jenkins-jobs/jenkins-jobs.mv.db``"). Support for an
+external H2 database and then for alternate databases (PostgreSQL)
+is on the roadmap
+(see https://github.com/jenkinsci/pipeline-maven-plugin/blob/pipeline-maven-3.0.0-beta-1/jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/dao/PipelineMavenPluginH2Dao.java[PipelineMavenPluginH2Dao.java])
```
```suggestion
[NOTE]
====
* The upstream artifact must be generated in a "[.code]``withMaven(){}``" wrapping step to be detected by the triggering system
* The downstream pipeline must have selected the build trigger "Build whenever a SNAPSHOT dependency is built"
 ** The build trigger can be defined at the pipeline level ("Build Triggers"), at the multibranch pipeline level ("Scan Repository Triggers") or at the GitHub Organization / Bitbucket Project level ("Scan Organizations Triggers")
* You have to manually trigger once the upstream pipeline and the downstream pipeline so that the link between the pipelines based on the SNAPSHOT dependency is established
* The dependency graph is, for the moment, exclusively stored in an H2 embedded database ("[.code]``$JENKINS_HOME/jenkins-jobs/jenkins-jobs.mv.db``"). Support for an external H2 database and then for alternate databases (PostgreSQL) is on the roadmap (see https://github.com/jenkinsci/pipeline-maven-plugin/blob/pipeline-maven-3.0.0-beta-1/jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/dao/PipelineMavenPluginH2Dao.java[PipelineMavenPluginH2Dao.java])
====
```

aheritier commented on README.adoc:
```diff
@@ -0,0 +1,471 @@
+Plugin]
+is installed)
+* image:docs/images/warning.svg[(warning)] Deprecation notice! Publish a report of the tasks ("[.code]``FIXME``" and "[.code]``TODO``") found in the
+java source code (if the https://wiki.jenkins-ci.org/display/JENKINS/Task+Scanner+Plugin[Jenkins Tasks Scanner
```
```suggestion
java source code (if the https://plugins.jenkins.io/tasks/[Jenkins Tasks Scanner
```

aheritier reviewed (approved):
It's awesome @bthompson0 
I proposed few edits to use admonitions (which could help to replace the panels in confluence) and the addition of a toc

I found the tasks Scanner plugin : https://plugins.jenkins.io/tasks/

I started to release a 3.9.0-beta-1 earlier today but I am blocked because of a permission issue.
As soon as it's done, I think we can merge your PR and release a beta 2 to see the result

bthompson0 commented on jenkins-plugin/src/resources/faq.md:
```diff
@@ -1,41 +0,0 @@
-
```
Oh yes, good point.

bthompson0 commented:
I made all the suggested changes locally and also added admonitions for the warnings too.

bthompson0 commented:
I have a couple of other documentation PRs on other Jenkins plugin repositories.  This was quite a learning experience for me.  Thanks for being patient and thorough on reviewing.  It was a pretty large README and FAQ.

I'll definitely apply what I learned to the other PRs.

aheritier commented:
perfect @bthompson0 
Thanks a lot !
